### PR TITLE
Show window message when auto extending import lists

### PR DESCRIPTION
**Pull request by berberman**
Imo we'd better notify users that we modified source code without their permission.

### CODE REVIEW

**Comment by pepeiborra** [comment]
I'm not sure this is a good idea. What does this look like in VSCode? And in other clients?

**Comment by berberman** [comment]
I tried with vscode and emacs:

### vscode
Peek

### emacs
Peek

**Review by pepeiborra — APPROVED** [review]

File: ghcide/src/Development/IDE/Plugin/Completions.hs [hunk 1]
```diff
@@ -159,9 +159,12 @@ extendImportCommand =
...
 
 extendImportHandler :: CommandFunction IdeState ExtendImport
-extendImportHandler ideState edit = do
+extendImportHandler ideState edit@ExtendImport {..} = do
   res <- liftIO $ runMaybeT $ extendImportHandler' ideState edit
-  whenJust res $ \wedit ->
+  whenJust res $ \wedit -> do
+    LSP.sendNotification SWindowShowMessage $
+      ShowMessageParams MtInfo $
+        "Import " <> maybe ("‘" <> newThing) (\x -> "‘" <> x <> " (" <> newThing <> ")") thingParent <> "’ from " <> importName
```
Comment (Ailrun) [inline ghcide/src/Development/IDE/Plugin/Completions.hs:167]: Do we have a starting line number information here? If so, I think it's better to print them too.

**Comment by berberman** [comment]
Perhaps it's a bit verbose to include the full file path?

Peek

**Review by Ailrun — APPROVED** [review]
Awesome, thanks!

**Comment by michaelpj** [comment]
It feels to me like this should be a client feature? The spec has `AnnotatedTextEdit`s that let you specify a human-readable name for the edit that is applied, including a part which should be "rendered prominent in the user interface" (https://microsoft.github.io/language-server-protocol/specification#textEdit). It seems like if we return `AnnotatedTextEdit`s with this information then it should be up to the client to show them to the user.

**Comment by michaelpj** [comment]
Looks like we maybe need support for them in the `lsp` library first, though..

**Pull request merged**